bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/scriptengine/DelegatingScriptEngineWithInvocableAndAutocloseable.java

```diff
@@ -33,94 +32,90 @@
  */
 public abstract class DelegatingScriptEngineWithInvocableAndAutocloseable<T extends ScriptEngine & Invocable & AutoCloseable>
         implements ScriptEngine, Invocable, AutoCloseable {
-    protected T delegate;
+    protected @NonNull T delegate;
```

Comment:
Why do you use `@NonNull`? Wouldn't it make sense to annotate the class with `@NonNullByDafault`? Are you running into trouble with inherited annotations? In that case ignore my comment, this can be fixed later.

Reply:
I tried `@NonNullByDefault` and it causes problems with inherited annotations.

Reply:
I'll split the null annotation changes to a separate PR to focus on the more important fix here.

Reply:
All fine with what we have. It's better than before and correct, as far as I can see.

Reply:
Okay, then I'll keep as it is.